webrtc.html

```diff
@@ -2033,6 +1684,537 @@
+            m-line in the remote description. This meant that at the caller,
+            MediaStreamTracks were not created until the answer was received, and any media
+            received prior to a remote description (AKA "early media") would be discarded.
+            If any form of remote description is provided (either an answer or a pranswer),
+            this issue does not occur.</p>
+
+            <p>If we want to allow early media to be played out, minor changes are necessary.
+            Fundamentally, we would need to change when tracks are created
+            for the offerer; this would have to happen either as a result of setLocalDescription,
+            or when media packets are received. This ensures that these objects can be created and
+            connected to media elements for playout.</p>
+
+            <p>However, there are three consequences to this potential change:
+              <ol>
+                <li>Media may arrive before the remote DTLS fingerprint has been received.
+                This means that media could be played out before the validity of the
+                DTLS fingerprint has been established, which may be hard to explain to users.
+                As such, it is recommended that media not be played out unless some
+                TBD RTCConfiguration property (e.g. AllowUnverifiedMedia) has been set.
+                </li>
+                <li>The information needed to correlate MediaStreamTracks with their
+                enclosing MediaStream will not yet be present when the tracks are
+                initially generated. Therefore, the implementation will need to
+                create dummy MediaStream objects for each MediaStreamTrack, and then
+                possibly change the associated MediaStream for each track when the
+                remote description is received (e.g. if it indicates that an audio
+                and video MediaStreamTrack should be combined into a single MediaStream).
+                Since media elements act on MediaStreams, some complex reshuffling may
+                need to occur when the remote description is received.
+                </li>
+                <li>The track events fired and their timing will change.
+                For the offerer, ontrack will now fire during setLocalDescription,
+                once for each track being offered, and track.onended will fire during
+                setRemoteDescription for any offered tracks that were rejected.
+                For the answerer, ontrack will continue to fire during setRemoteDescription,
+                as it does today (this is necessary to allow the answerer to reject
+                offered tracks by stopping them).
+                </li>
+              </ol>
+            </p>
+
+            <p>For now, we simply make note of this issue, until it can be
+            considered fully by the WG.</p>
```

Comment:
Cullen, you have repeatedly been asked to make suggestions on how to deal with early media.
My preference is to just bit-bucket packets before first answer; if an application knows it's going to interwork with legacy only, it can generate a pr-answer with no candidates and no msid-semantic, and tracks will pop up per the -msid document.
I've yet to see a case that is compelling to me where that workaround is not sufficient.